app/routes.py

```diff
@@ -1,2 +1,231 @@
-from flask import Blueprint
+from flask import Blueprint, request, make_response, jsonify
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import datetime
+import os
+from dotenv import load_dotenv
 
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+
+@tasks_bp.route("", methods=["POST","GET"])
+def handle_tasks():
+    if request.method=="POST":
+        request_body = request.get_json()
+        if "completed_at" not in request_body.keys() or "description" not in request_body.keys() or "title" not in request_body.keys():
+            return make_response({"details": "Invalid data"}, 400) 
+        else:
+            new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+            db.session.add(new_task)
+            db.session.commit()
+            return make_response({"task": {
+                    "id": new_task.task_id,
+                    "title": new_task.title,
+                    "description": new_task.description,
+                    "is_complete": False if new_task.completed_at is None else True
+                }}, 201)
+
+    elif request.method=="GET":
+        title_query = request.args.get('sort')
+        if title_query=='desc':
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        elif title_query=='asc':
+            tasks = Task.query.order_by(Task.title).all()
+        else:
+            tasks=Task.query.all()
+        tasks_response=[]
+        for task in tasks:
+            tasks_response.append({
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at is None else task.completed_at
+                })
+        return jsonify(tasks_response)
+
+    
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_a_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response('',404)
+
+    elif request.method == "GET":
+        if not task.goal_id:
+            return {"task": { "id": task.task_id,
+                        "title": task.title,
+                        "description": task.description,
+                        "is_complete": False if task.completed_at is None else task.completed_at  
+                        }}
+        else:
+            return {"task": { "id": task.task_id,
+                        "title": task.title,
+                        "description": task.description,
+                        "goal_id":task.goal_id,
+                        "is_complete": False if task.completed_at is None else task.completed_at  
+                        }}
```

Comment:
Like I mentioned above you can move this to your task model and create an instance method that could be called here instead of repeating your code.